model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java

```diff
@@ -1050,7 +1050,8 @@ public void setDefaultRole(RoleModel role) {
 
     @Override
     public RoleModel getDefaultRole() {
-        return cached.getDefaultRoleId() == null ? null : cacheSession.getRoleById(this, cached.getDefaultRoleId());
+        if (isUpdated()) updated.getDefaultRole();
```

Comment:
Should this be the following?
```suggestion
        if (isUpdated()) {
            return updated.getDefaultRole();
        }
```